src/Sylius/Bundle/ApiBundle/Doctrine/QueryItemExtension/OrderMethodsItemExtension.php

```diff
@@ -57,15 +57,36 @@ public function applyToItem(
         $rootAlias = $queryBuilder->getRootAliases()[0];
         $user = $this->userContext->getUser();
 
-        $this->applyUserRulesToItem($user, $queryBuilder, $rootAlias, $httpRequestMethodType);
+        $this->applyUserRulesToItem($user, $queryBuilder, $rootAlias, $httpRequestMethodType, $operationName);
     }
 
     private function applyUserRulesToItem(
         ?UserInterface $user,
         QueryBuilder $queryBuilder,
         string $rootAlias,
-        string $httpRequestMethodType
+        string $httpRequestMethodType,
+        string $operationName
     ): void {
+        if ($user instanceof ShopUserInterface && $operationName === 'shop_select_payment_method') {
```

Comment:
I suggest refactoring the existing code to apply on query builder
```
->andWhere(sprintf('%s.state = :state', $rootAlias))
->setParameter('state', OrderInterface::STATE_CART)
```
only if there is an endpoint for checkout.
IMO it could simplify the logic 😃